pilot/pkg/config/monitor/copilot_snapshot.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitor
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	copilotapi "code.cloudfoundry.org/copilot/api"
+
+	networking "istio.io/api/networking/v1alpha3"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/log"
+)
+
+// CopilotClient defines a local interface for interacting with Cloud Foundry Copilot
+//go:generate counterfeiter -o fakes/copilot_client.go --fake-name CopilotClient . CopilotClient
```

Comment:
Just to clarify: would you like us to add `counterfeiter` to the vendor so that we can `go run` it, kinda like https://github.com/istio/istio/blob/fae56a635620758ea178301c176de2bc9fb262b9/mixer/template/sample/doc.go#L31

Reply:
discussed in slack, we are leaving this as is.

Reply:
We chatted on Slack: I thought `go run` would download the binary if provided a fully qualified name; it does not. So this is fine. If we hit this footgun down the line and have issues due to the binary missing, we can vendor it. The PR can go in with this as-is though.